packages/core/content-manager/server/src/controllers/relations.ts

```diff
@@ -458,7 +458,7 @@ export default {
       ordering: 'desc',
     });
 
-    const relationsUnion = uniqBy('id', concat(sanitizedRes.results, res.results));
+    const relationsUnion = uniqBy('documentId', concat(sanitizedRes.results, res.results));
```

Comment:
I wanted to make this change last month but Marc warned me it could break some scenarios. Could you check [his comments in that thread](https://github.com/strapi/strapi/pull/21010/files/f8c21235d3e01a5fa5b2dd7753ea53d7eb395bb7#r1716846564)? I don't know if they're use cases we want to support

Reply:
Nice catch 🎉  I updated the condition to check by docId and locale as we need the status to be ignored